// Copyright 2020 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// This proto library is only used to define various messages for use by the
// unit tests in transform_test.cc. The structures defined here are not (and
// should not) be published outside of that context; this also means that we are
// free to change these definitions without concern for backwards compatibility.

syntax = "proto3";

package ecclesia_transform_test;

message SimpleMessage {
  string devpath = 1;
  string not_devpath = 2;
  string also_devpath = 3;
  int32 not_even_a_string = 4;
}

message NestedMessage {
  message Submessage {
    string devpath = 1;
    int32 number = 2;
  }
  Submessage sub = 1;
}

message RepeatedMessage {
  repeated string multiple_devpaths = 1;
}

message ComplicatedMessage {
  message B {
    message C {
      string devpath = 1;
    }
    repeated string multiple_devpaths = 1;
    repeated C cc = 2;
  }
  B bb = 1;
  string also_devpath = 3;
}

message OneofMessage {
  oneof o1 {
    string devpath = 1;
    int32 not_devpath = 2;
    string also_devpath = 3;
  }
  message A {
    string devpath = 1;
  }
  oneof o2 {
    A a = 4;
    int32 not_devpath2 = 5;
  }
  message B {
    repeated string devpath = 1;
  }
  oneof o3 {
    B b = 6;
    int32 not_devpath3 = 7;
  }

  message C {
    oneof o {
      string devpath = 1;
      int32 not_devpath = 2;
    }
  }
  oneof o4 {
    C c = 8;
    int32 not_devpath4 = 9;
  }

  oneof o5 {
    string another_devpath = 10;
    int32 another_not_devpath = 11;
  }
}

message NestedRepeatedOneof {
  message A {
    string devpath_a = 1;
  }
  message B {
    string devpath_b = 1;
  }
  message Repeated {
    oneof o {
      A sub_a = 1;
      B sub_b = 2;
    }
  }
  repeated Repeated r = 1;
}
